eclair-core/src/test/scala/fr/acinq/eclair/payment/PostRestartHtlcCleanerSpec.scala

```diff
@@ -419,6 +421,45 @@ class PostRestartHtlcCleanerSpec extends TestKitBaseClass with FixtureAnyFunSuit
     channel_upstream_3.expectNoMsg(100 millis)
   }
 
+  test("handle a channel relay htlc-fail") { f =>
+    import f._
+
+    val testCase = setupChannelRelayedPayments(nodeParams)
+    val relayer = f.createRelayer()
+    register.expectNoMsg(100 millis)
+
+    // we need a reference to the post-htlc-restart child actor
+    sender.send(relayer, Relayer.GetChildActors(sender.ref))
+    val postRestartHtlcCleaner = sender.expectMsgType[Relayer.ChildActors].postRestartCleaner
+
+    sender.send(relayer, buildForwardFail(testCase.downstream, testCase.origin))
+    register.expectMsgType[Register.Forward[CMD_FAIL_HTLC]]
+
+    register.expectNoMsg(100 millis) // the payment has already been fulfilled upstream
+    eventListener.expectNoMsg(100 millis)
+  }
+
+  test("handle a channel relay htlc-fulfill") { f =>
+    import f._
+
+    val testCase = setupChannelRelayedPayments(nodeParams)
+    val relayer = f.createRelayer()
+    register.expectNoMsg(100 millis)
+
+    // we need a reference to the post-htlc-restart child actor
+    sender.send(relayer, Relayer.GetChildActors(sender.ref))
+    val postRestartHtlcCleaner = sender.expectMsgType[Relayer.ChildActors].postRestartCleaner
+
+    // This downstream HTLC has two upstream HTLCs.
+    sender.send(relayer, buildForwardFulfill(testCase.downstream, testCase.origin, preimage1))
+    register.expectMsg(Register.Forward(postRestartHtlcCleaner, testCase.origin.originChannelId, CMD_FULFILL_HTLC(testCase.origin.originHtlcId, preimage1, commit = true)))
+
+    sender.send(relayer, buildForwardFulfill(testCase.downstream, testCase.origin, preimage1))
+    register.expectNoMsg(100 millis) // the payment has already been fulfilled upstream
+    eventListener.expectNoMsg(100 millis)
```

Comment:
We should expect a `ChannelPaymentRelayed` (once implemented of course)

Reply:
Done with 6be5ba134.